### ISSUE TITLE
Cannot override built-in connection variable ansible_become_pass

### ISSUE THREAD

**Issue by patran** [ansible/ansible#21639]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

Cannot declare ansible_become_pass in variable declaration in playbook "vars". 

##### ANSIBLE VERSION

```
ansible 2.2.1.0

  config file = /myprojects/playbook-abc/ansible.cfg
  configured module search path = Default w/o overrides
```

##### OS / ENVIRONMENT

Ubuntu 16.04.2 using virtualenv

##### SUMMARY

In the playbook vars section, the following produces a runtime error:
  vars:
    ansible_become_pass: "vault_{{ansible_user}}_ansible_become_pass"

##### STEPS TO REPRODUCE

```yaml
---
---
- hosts: all
  gather_facts: no

  vars:
    user_become_pass: "vault_{{ansible_user}}_ansible_become_pass"
    ansible_become_pass: "vault_{{ansible_user}}_ansible_become_pass"
     
  tasks:
  - name: a
    debug:
      msg: "{{user_become_pass}}"
      
  - name: b
    debug:
      msg: "{{ansible_become_pass}}"
```

##### EXPECTED RESULTS

Change ansible_become_pass to dansible_become_pass and the playbook would run just fine
Note: ansible_become_pass is a built-in connection variable, there seems to be naming conflict somehow. Hence, prefixing with a 'd' solved the issue at hand. However, I do want to be able to override ansible_become_pass

##### ACTUAL RESULTS

The playbook does not run

```
fatal: [192.168.122.86]: FAILED! => {
    "failed": true,
    "msg": "the field 'become_pass' has an invalid value, which appears to include a variable that is undefined. The error was: 'ansible_user' is undefined"
```

**Comment by alikins** [comment]
The 'ansible_user' variable will only be defined if a remote user is specified ('ansible-playbook -u username' or remote_user in inventory. See https://github.com/ansible/ansible/issues/14195 for more details).

So if ansible_user is undefined, any variables that reference it are also undefined.

**Comment by patran** [comment]
If you rename ansible_user to "dansible_user", it will work. I don't think it is related to "-u"

**Comment by bcoca** [comment]
so #38861 should fix this issue by pushing evaluation of vars down to the new become plugins and avoiding the play_context having a value before task evaluation.

**Comment by samdoran** [comment]
Closing since this was resolved by #38861.
